src/pykeen/nn/representation.py

```diff
@@ -1137,7 +1139,28 @@ def _plain_forward(
         return self.combine(combination=self.combination, base=self.base, indices=indices)
 
 
-class WikidataTextRepresentation(TextRepresentation):
+class CachedTextRepresentation(TextRepresentation):
+    """Textual representations for datasets with identifiers that can be looked up with a :class:`TextCache`."""
+
+    cache_cls: Type[TextCache]
+
+    def __init__(self, labels: Sequence[str], **kwargs):
```

Comment:
Since "labels" and "ids" are a bit ambiguous here, maybe we should be more verbose in the parameter name, e.g.,
```suggestion
    def __init__(self, textual_identifiers: Sequence[str], **kwargs):
```